### ISSUE TITLE
1.17 compatibility

### ISSUE THREAD

**Issue by angelandasi** [clonemeagain/plugin-autocloser#44]
Hi,
I just upgraded to the version 1.17 of osTicket. Since then, I am getting DB Error #1062 in osTicket logs every 5 minutes.
The error shows autocloser plugin folder and class.CloserPlugin.php. So I guess it is related to this plugin. I disabled the plugin and I don't get the error log anamore.

Does anyone know what I can do to make it work again?
Thank you in advance.

Our osTicket settings:

osTicket Version | v1.17 (1d8b790) —  Up to date
Web Server Software | Apache/2.4.37 () OpenSSL/1.1.1g Phusion_Passenger/6.0.6
MySQL Version | 5.7.34
PHP Version | 8.0.11

This is the entire log message I am getting:

DB Error #1062
[INSERT INTO `hgjv_config` SET `key` = 'last-run', `value` = 1666085703, `updated` = NOW()] Duplicate entry 'last-run' for key 'namespace'

 ---- Backtrace ----
 #0 (root)/include/mysqli.php(203): osTicket->logDBError('DB Error #1062', '[INSERT INTO h...')
 #1 (root)/include/class.orm.php(3482): db_query('INSERT INTO `hg...', true, true)
 #2 (root)/include/class.orm.php(658): MySqlExecutor->execute()
 #3 (root)/include/class.config.php(176): VerySimpleModel->save(true)
 #4 (root)/include/class.config.php(109): ConfigItem->save()
 #5 (root)/include/class.config.php(119): Config->create('last-run', 1666085703)
 #6 (root)/include/class.config.php(84): Config->update('last-run', 1666085703)
 #7 (root)/include/plugins/autocloser/class.CloserPlugin.php(203): Config->set('last-run', 1666085703)
 #8 (root)/include/plugins/autocloser/class.CloserPlugin.php(85): CloserPlugin->is_time_to_run(Object(CloserPluginConfig))
 #9 (root)/include/plugins/autocloser/class.CloserPlugin.php(68): CloserPlugin->logans_run_mode()
 #10 (root)/include/class.signal.php(98): CloserPlugin->{closure}(NULL, Array)
 #11 (root)/include/class.cron.php(123): Signal::send('cron', NULL, Array)
 #12 (root)/include/api.cron.php(19): Cron::run()
 #13 (root)/include/api.cron.php(40): CronApiController->run()
 #14 (root)/api/cron.php(23): LocalCronApiController::call()
 #15 {main}

**Comment by pierromanu** [comment]
I'm facing the same issue:

[INSERT INTO `config` SET `key` = 'last-run', `value` = 1666195289, `updated` = NOW()] Duplicate entry 'last-run' for key 'namespace'<br /> <br /> ---- Backtrace ----<br /> #0 (root)/include/mysqli.php(203): osTicket->logDBError()<br /> #1 (root)/include/class.orm.php(3482): db_query()<br /> #2 (root)/include/class.orm.php(658): MySqlExecutor->execute()<br /> #3 (root)/include/class.config.php(176): VerySimpleModel->save()<br /> #4 (root)/include/class.config.php(109): ConfigItem->save()<br /> #5 (root)/include/class.config.php(119): Config->create()<br /> #6 (root)/include/class.config.php(84): Config->update()<br /> #7 (root)/include/plugins/plugin-autocloser-master/class.CloserPlugin.php(188): Config->set()<br /> #8 (root)/include/plugins/plugin-autocloser-master/class.CloserPlugin.php(79): CloserPlugin->is_time_to_run()<br /> #9 (root)/include/plugins/plugin-autocloser-master/class.CloserPlugin.php(68): CloserPlugin->logans_run_mode()<br /> #10 (root)/include/class.signal.php(98): CloserPlugin->{closure}()<br /> #11 (root)/include/class.cron.php(123): Signal::send()<br /> #12 (root)/include/api.cron.php(19): Cron::run()<br /> #13 (root)/include/api.cron.php(40): CronApiController->run()<br /> #14 (root)/api/cron.php(23): LocalCronApiController::call()<br /> #15 {main

Please let me know your findings

**Comment by extent-ops** [comment]
https://github.com/clonemeagain/plugin-autocloser/issues/45

**Comment by angelandasi** [comment]
Thank you @extent-ops. It worked for me after following your instructions on #45 issue.
It was a bit confusing for me due to all the different message, so I decide to add here wath I changed.
This is what I changed (in yellow):

Near line 54
![image](https://user-images.githubusercontent.com/99033842/198246571-8d58edf2-f8ef-474b-afa8-ba491a8fec16.png)

and near line 81
![image](https://user-images.githubusercontent.com/99033842/198247472-83272264-ca6c-40a3-873f-8290ef161d36.png)

and near line 200
![image](https://user-images.githubusercontent.com/99033842/198247771-91cbab17-c56b-4a75-8b3a-e0cddfb3b093.png)
